File: android/SwiftKotlination/app/src/sharedTest/kotlin/fr/jhandguy/swiftkotlination/network/File.kt
package fr.jhandguy.swiftkotlination.network

import java.io.InputStream

class File(name: String, extension: Extension) {
    enum class Extension {
        JSON, JPG;
        var value: String = name.toLowerCase()
    }

    val data: InputStream? = javaClass.classLoader?.getResourceAsStream("$name.${extension.value}")
}

File: ios/NetworkKit/NetworkKit/Sources/Stubs/File.swift
import Foundation

public final class File: Codable {
    var name: String
    var `extension`: Extension

    public enum Extension: String, Codable {
        case json
        case jpg
    }

    public init(_ name: String, _ extension: Extension) {
        self.name = name
        self.extension = `extension`
    }

    public var data: Data? {
        guard
            let url = Bundle(for: type(of: self)).url(forResource: name, withExtension: `extension`.rawValue),
            let data = try? Data(contentsOf: url) else {
            return nil
        }

        return data
    }
}
